### PR TITLE
DPTS-683 Scenario upload lambda

### PR DESCRIPTION
**Pull request by timrickwood**

### CODE REVIEW

File: application/hk/stt/handler.py [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+        bundle = common.retrieve_compressed_file_from_bucket(bucket, filename, event, start)
+        logger.log_for_audit(env, "action:bundle {} downloaded".format(filename))
+        bundle_id = add_bundle(env, db_connection, filename)
+        processed = process_zipfile(env, db_connection, bundle, filename, bundle_id, scenario_count)
+        common.report_summary_counts(scenario_count, env)
+        if processed is True:
+            message.send_success_slack_message(event, start, scenario_count)
+        else:
+            message.send_failure_slack_message(event, start)
+        # TODO will need to unpack
```
Comment (timrickwood) [inline application/hk/stt/handler.py:41]: Remove TODO

File: application/hk/stt/handler.py [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+    message.send_start_message(event, start)
+    filename = event["filename"]
+    bucket = event["bucket"]
+    env = event["env"]
+    db_connection = None
+    logger.log_for_audit(event["env"], "action:task started")
+    scenario_count = initialise_count()
+    try:
+        db_connection = database.connect_to_database(env)
+        # TODO will be a compressed file - testing on .zip -  rar?
```
Comment (timrickwood) [inline application/hk/stt/handler.py:31]: Remove TODO

File: application/utilities/common.py [hunk 3]
```diff
@@ -123,15 +129,28 @@ def report_summary_counts(summary_count_dict, env):
     log_for_audit(env, slack_summary_counts(summary_count_dict))
 
 
+# def slack_summary_counts_old(summary_count_dict):
```
Comment (timrickwood) [inline application/utilities/common.py:132]: Remove commented out code

File: test/unit/utilities/test_utilities_s3.py [hunk 4]
```diff
@@ -13,6 +15,43 @@
...
+    s3_client.upload_file(Filename=sample_bundle_file_name, Bucket=mock_bucket, Key=mock_zip_filename)
+    response_body = S3().get_compressed_object(mock_bucket, mock_zip_filename, mock_event, start)
+    # print (response_body)
+    # assert True == False
+    # contents = response["Body"].read()
+    # assert zipfile.is_zipfile(response_body)
+    input_zip = zipfile.ZipFile(io.BytesIO(response_body))
+
+    for name in input_zip.namelist():
+        # print("=========== " + name + "============")
```
Comment (timrickwood) [inline test/unit/utilities/test_utilities_s3.py:36]: Remove commented out code

File: application/hk/stt/handler.py [hunk 5]
```diff
@@ -0,0 +1,346 @@
+import xml.etree.ElementTree as ET
+import zipfile
+import io
+import xmltodict
+
+#  works but not from docker
```
Comment (timrickwood) [inline application/hk/stt/handler.py:6]: Remove commented out code

File: build/automation/var/profile/nonprod.mk [hunk 6]
```diff
@@ -14,7 +14,7 @@ VULNERABILITY_LEVEL := CRITICAL,HIGH
 
 STACKS := secrets,s3,security-groups
 # TODO restore  referralroles - temp out to avoid cross-cutting other work
```
Comment (timrickwood) [inline build/automation/var/profile/nonprod.mk:16]: Remove commented out code

**Review by tamaragoldschmidt — APPROVED** [review]

**Pull request merged**